### PR TITLE
Features/5 manage trips

### LINKED ISSUE #5: Gestion des voyages
## User stories
A travers l'API, permette de crée un voyage, de le modifier, de le supprimer ainsi que de pouvoir avoir accès à un ou plusieurs voyages.

## Description of needs
- End point POST création de voyage
- _Les Put et patch nous verrons au besoin, plus en mode features. A première vus il n'y aurais que du Patch_
- End point DELETE suppression de voyage
- End point GET récupération de un voyage
- End point GET récupération de voyages by user

## Acceptance criteria (done when...)

- [ ] Chaque End point a son ou ces tests unitaires
- [x] Chaque End Point a son validator
- [x] Chaque End Point est mapper vers un DTO
- [x] Chaque End Point est documenter

### Ajout d'un voyage 
- [x] DTO Ajout de voyage
- [x] Validator de ce DTO
- [x] Mapper DTO to Entity
- [x] Service d'ajout du voyage
- [x] Repository d'ajout du voyage
### Suppression d'un voyage 
- [x] DTO Delete de voyage
- [x] Validator Du DTO
- [x] Service de delete du voyage
- [x] Repository de delete du voyage
### Récupération d'un seul voyage
- [x] Route contenant l'id du voyage
- [x] Service de récupération du voyage
- [x] Repository de récupération du voyage
- [x] Mapper de l'entity vers DTO
- [x] DTO de voyage _Adapté à l'affichage d'un seul voyage, donne un max de details possible. Doit être évolutif pour prendre en compte par la suite les éléments qui peuplerons les voyages_
### Récupération des voyages d'un utilisateur
- [x] Route contenant l'id de l'utilisateur
- [x] Service de récupération des voyages
- [x] Repository de récupération des voyages
- [x] Mapper des entities vers la liste de DTO
- [x] DTO de voyages _Adapté à l'affichage d'une liste des voyages, titre, desc, id, par exemple_

### PR DESCRIPTION
**Pull request by Dercraker**
- closes #5

### CODE REVIEW

File: .vscode/icon.code-snippets
Comment (Baptiste-Ferrand) [inline .vscode/icon.code-snippets:1]: u want to push ur .vscode?

File: src/components/trip/createTripModal.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                )
+              }
+              {...createTripForm.getInputProps('description')}
+            />
+            <DatePickerInput
+              withAsterisk
+              label="Departure date"
+              description="Enter your departure date"
+              placeholder={moment().format('YYYY, MMMM D')}
+              valueFormat="YYYY, MMMM D"
```
Comment (Baptiste-Ferrand) [inline src/components/trip/createTripModal.tsx:128]: maybe beginning is more appropriate.
is the beginning to your trips 
is the departure to your fly

File: .vscode/icon.code-snippets
Comment (Dercraker) [reply .vscode/icon.code-snippets:1]: Yeah, it's voluntary. They're just snippets that make it easier for the whole team to dev.

File: src/components/trip/createTripModal.tsx (hunk 1 above)
Comment (Dercraker) [reply src/components/trip/createTripModal.tsx:128]: it's right, i will be edit this

**Review by BaptisteLegat — APPROVED** [review]

**Review by Baptiste-Ferrand — APPROVED** [review]

**Pull request merged**